### DISCUSSION TITLE
Unfamiliar Styling System

### DISCUSSION THREAD

**Discussion by bidrang** [tamagui/tamagui#418] [General]
Hi there,
First of all, thanks for the great job you are doing. I believe it would have a very bright future.

One thing I thought it's essential to have before launching the first release is making the Tamagui styling system more familiar with the other known style systems.

for instance, I believe Chakra-UI sizes are easier for designers and developers to understand and remember:
https://chakra-ui.com/docs/styled-system/theme#spacing

Best,
Saeed

**Comment by jonsherrard** [comment]
I know you're coming from, I miss a few things coming from the `styled-ui`, but given Nate's expertise and vision - I've been letting go of my habits and trying to adopt a "Tamagui" approach. 

All in all it's very flexible, with the config system and component variants, you utliise many of the configurations and APIs you might use in other systems. 

If you want, I'm _pretty_ sure you can copy and paste, the spacing system from Chakra for example, (maybe with a `$` prefix on some properties) and you can use that in your Tamagui Config, using the `createTokens` method.

**Comment by bidrang** [comment]
@jonsherrard I love what Tamagui has done for making the configuration flexible, and I know there is a possibility to define our own spacing variables there. But I believe if Tamagui sticks to the design systems which is already trusted by the other developers and designers and used in popular systems like Tailwind and Chakra, it will bring better DX for Tamagui and make it easier to migrate to this great product.

I believe it is just the beginning of the Tamagui journey and if @natew may agree with what I noted here, it's better to make this "Breaking Change" before the first official release. And for sure in this case for the users which already using the Tamagui, there could be a config file to override this change.

**Reply by redbar0n** [reply]
Show demand for Tailwind-like API support here: https://github.com/tamagui/tamagui/discussions/114

**Reply by bidrang** [reply]
I believe that to provide a Tailwind-like API, which I agree with it; first of all, Tamagui needs to make spacing, sizes, and other variables and names compatible with Tailwind (as I mentioned above). Otherwise, it will be difficult for designers and developers to switch to Tamagui and to grow its eco-system.

**Reply by jonsherrard** [reply]
Maybe you would like https://www.nativewind.dev and the Solito project that lets you write TW in React Native and the Web?  https://www.nativewind.dev/quick-starts/solito

**Reply by bidrang** [reply]
I have seen that project as well which is also in heavy development. for me, it's a kind of comparison between Tailwind and Chakra but for developing universal apps.
To be honest, as a decision maker in my company at the moment I still prefer to use Tailwind and its third-party UI-Kits over Chakra-UI and its great design system and components libraries, as the whole team is familiar with that and there are lots of resources over the Internet. 
But if Tamagui becomes the Chakra replacement for universal apps, for sure it will change the whole game!
